lib/dpl/provider/pages.rb

```diff
@@ -44,34 +53,140 @@ def slug
         options.fetch(:repo) { context.env['TRAVIS_REPO_SLUG'] }
       end
 
+      def keep_history
+        options.fetch(:keep_history) { false }
```

Comment:
https://ruby-doc.org/core-2.2.0/Hash.html#method-i-fetch suggests you can write this as `options.fetch(:keep_history, false)` instead of using a block.